lib/main.js

```diff
@@ -189,6 +200,21 @@ export default {
     });
   },
 
+  async getProgram(Linter, configurationPath) {
+    let program;
+    const tsconfigPath = path.resolve(path.dirname(configurationPath), 'tsconfig.json');
```

Comment:
Instead of trying to check if `tsconfig.json` exists under `configurationPath`, I would use [`Linter.findConfigurationPath(null, configurationPath)`](https://github.com/palantir/tslint/blob/master/src/configuration.ts#L116) to fetch the config file, this way it will call TSLint's logic for searching up the hierarchy; and we get the search code for free (can remove the [`statAsync()`](https://github.com/AtomLinter/linter-tslint/pull/157/files#diff-16b7d75b70953cbf4755508170f2f24cR16) method you added)